FRES-265: Add custom module.

Aryan-Singh-Innoraft opened the pull request.

Aryan-Singh-Innoraft committed:
FRES-265: Add custom module.

ankitdebnath003 reviewed (changes requested):
1. Did you enable this module?
2. Which role has the custom permission to view this route?

Aryan-Singh-Innoraft commented on web/modules/custom/welcome_user/src/Controller/WelcomeUserController.php:
```diff
@@ -0,0 +1,27 @@
+
+    return[
+      '#markup' => $this->t('Hello @name' , ['@name' => $username]),
+      '#cache' => ['max-age' => 0,],
```
max-age 0 tells drupal not to cache this content as I want it to be rebuild everytime the user visits the page.

ankitdebnath003 replied:
Is it optimized way to do this?

Aryan-Singh-Innoraft replied:
Updated that part

Aryan-Singh-Innoraft commented:
> 1. Did you enable this module?
> 2. Which role has the custom permission to view this route?

1.Yes I have enabled this module.
2.The roles that have permission to view this route are-
Anonymous user
Administrator
Content Editor

Aryan-Singh-Innoraft committed:
FRES-265: Update the cache part in Controller.

Aryan-Singh-Innoraft committed:
FRES-265: Add the config files.

ankitdebnath003 reviewed (changes requested):
1. Why not providing permission to Authenticated user role?
2. Why so many modules are there in core.extension except the one created for this ticket?

Aryan-Singh-Innoraft committed:
FRES-265: Add permission to authenticated user.

Aryan-Singh-Innoraft commented:
> 1. Why not providing permission to Authenticated user role?
> 2. Why so many modules are there in core.extension except the one created for this ticket?



> 1. Why not providing permission to Authenticated user role?
> 2. Why so many modules are there in core.extension except the one created for this ticket?

1.Yes I have added permission for the authenticated user and added the file as well.
2.The other modules are also visible because I have installed it for the other branches so all the modules are being shown when I am exporting.
I thought that the configuration is getting saved so I thought that I will have to uninstall all the other modules from the site,but for that I will have to switch to all the other branches and uninstall the modules individually.

Aryan-Singh-Innoraft committed:
FRES-265: Add core extension.

Aryan-Singh-Innoraft commented:
> > 1. Why not providing permission to Authenticated user role?
> > 2. Why so many modules are there in core.extension except the one created for this ticket?
> 
> > 1. Why not providing permission to Authenticated user role?
> > 2. Why so many modules are there in core.extension except the one created for this ticket?
> 
> 1.Yes I have added permission for the authenticated user and added the file as well. 
2.The other modules are also visible because I have installed it for the other branches so all the modules are being shown when I am exporting. I thought that the configuration is getting saved so I thought that I will have to uninstall all the other modules from the site,but for that I will have to switch to all the other branches and uninstall the modules individually.
I have solved this issue by switching to that branch and running cim command which solves the problem.I have done the new commit also.

ankitdebnath003 reviewed (changes requested):
Where is the info.yml file for this module?

Aryan-Singh-Innoraft committed:
FRES-265:Fix the gap issue.

Aryan-Singh-Innoraft committed:
FRES-265: Add comments and fix gap issue.

Aryan-Singh-Innoraft committed:
FRES-265: Fix gap issue.

Aryan-Singh-Innoraft committed:
FRES-265: Solve gap issue.

Aryan-Singh-Innoraft committed:
FRES-265: Merge changes from main.
